spec/contracts/rspec/mocks_spec.rb

```diff
@@ -1,48 +1,56 @@
-class Something
+class Plant
   include Contracts
 
-  Contract None => :something_is_done
+  Contract None => :success
   def call
-    :something_is_done
+    :success
   end
 end
 
+class Tree < Plant
+end
+
+class Oak < Tree
+end
+
 class Example
   include Contracts
 
-  Contract Something => :something_is_done
-  def do_something(something)
-    something.call
-  end
-
-  Contract String => Something
-  def returns_something
-    Something.new
+  Contract Plant => :success
+  def test(instance)
+    instance.call
   end
 end
 
 RSpec.describe Example do
-  let(:something) { instance_double(Something, call: :something_is_done) }
+  let(:plant) { instance_double(Plant, call: :success) }
+  let(:tree) { instance_double(Tree, call: :success) }
+  let(:oak) { instance_double(Oak, call: :success) }
 
   context "without Contracts::RSpec::Mocks included" do
     it "violates contract" do
-      expect { subject.do_something(something) }
-        .to raise_error(ContractError, /Expected: Something/)
+      expect { subject.test(plant) }
+        .to raise_error(ContractError, /Expected: Plant/)
     end
   end
 
   context "with Contracts::RSpec::Mocks included" do
     include Contracts::RSpec::Mocks
 
     it "succeeds contract" do
-      expect(subject.do_something(something)).to eq(:something_is_done)
+      expect(subject.test(plant)).to eq(:success)
     end
 
-    it "violates return value contract" do
-      something = instance_double(Something)
-      allow(Something).to receive(:new).and_return(something)
+    context "when doubled class is a subclass of contract class" do
+      it "succeeds contract" do
+        expect(subject.test(tree)).to eq(:success)
+      end
+    end
 
-      Example.new.returns_something
+    context "when doubled class is a descendant of contract class" do
+      it "succeeds contract" do
+        expect(subject.test(oak)).to eq(:success)
+      end
```

Comment:
Can you add one more test with a failure example for when the instance passed is a class `Plant` inherits from? 

Like maybe:
```
class Plant < Eukaryote
...
end

context "when doubled class is a parent class of contract class" do
  it "violates contract" do
    ...
  end
end
```

Reply:
Good call. What do you think of this? https://github.com/PrintReleaf/contracts-rspec/pull/2/commits/77e17c5a0732bec5f269033d98da2c0fdf0990ad

Also, is my taxonomy correct? (class hierarchies are hard)

Reply:
Looks good and is correct (I def googled eukaryote for my example because I couldn't remember either)